src/gwt/src/org/rstudio/studio/client/workbench/snippets/SnippetHelper.java

```diff
@@ -280,7 +281,7 @@ private String transformMacros(
          snippet = replaceHeaderGuard(snippet);
       }
 
-      return snippet.replaceAll("\\$\\$", token.substring(snippetName.length()));
+      return snippet.replaceAll("(?<!\\\\)\\$\\$", token.substring(snippetName.length()));
```

Comment:
IIUC replacing `$$` in the snippets supports this:  https://support.rstudio.com/hc/en-us/articles/204463668-Code-Snippets?version=2022.09.999-dev%2B999&mode=server

<img width="652" alt="image" src="https://user-images.githubusercontent.com/2625526/190429947-753e8816-06a1-482e-871d-19b32cd0e2bf.png">

so I believe the workaround is ok.